src/ring_theory/polynomial/eisenstein.lean

```diff
@@ -242,6 +244,56 @@ lemma cyclotomic_comp_X_add_one_is_eisenstein_at [hp : fact p.prime] :
     exact nat.prime.not_dvd_one hp.out (dvd.intro k (hk.symm)),
   end }
 
+lemma cyclotomic_prime_pow_comp_X_add_one_is_eisenstein_at [hp : fact p.prime] (n : ℕ) :
+  ((cyclotomic (p ^ (n + 1)) ℤ).comp (X + 1)).is_eisenstein_at 𝓟 :=
+{ leading :=
+  begin
+    rw [show (X + 1 : ℤ[X]) = X + C 1, by simp],
+    refine ((cyclotomic.monic _ ℤ).comp (monic_X_add_C 1) (λ h, _)).leading_coeff_not_mem
+      (ideal.is_prime.ne_top $ (ideal.span_singleton_prime (by exact_mod_cast hp.out.ne_zero)).2
+      $ nat.prime_iff_prime_int.1 hp.out),
+    rw [nat_degree_X_add_C] at h,
```

Comment:
I don't know why but I didn't manage to use `simpa` to close the goal in one line.